lib/rrd_fring.mli

```diff
@@ -11,21 +11,29 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
  *)
-(** Ring structures *)
- 
+(** Ring structures for RRAs
+    The values in the structures are bound to a range *)
+
+module BoundedFloat = Rrd_utils.BoundedFloat
+
 type t = {
   size : int;
   mutable current : int;
+  min : float;
+  max : float;
```

Comment:
Ah I see  you cannot change the type from float to BoundedFloat.t because it is inside a Bigarray.
Perhaps we should instead have a BoundedFloatBigarray.t, which doesn't expose the bigarray directly, but just a set operation that does the clamping, etc., and a get function that reads it.
This'd also solve your other problem with the temporary bigarray: no temporaries would be needed: you could read back the value you've written and do the clamping again if needed.

As it is written currently it is still possible to get access to `data` directly and change it without going through BoundedFloat.

Reply:
I've made `Fring.t` private, the only method that writes to `data` is `Fring.push`, which does the clamping. This should ensure all values in the bigarrays are clamped.